File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if not head or not head.next or k==0:
            return head
        temp=head
        n=0
        while temp:
            n=n+1
            temp=temp.next
        k=k%n
        for i in range(k):
            temp=head
            prev=None
            while temp.next:
                prev=temp
                temp=temp.next
            prev.next=None
            temp.next=head
            head=temp
        return head



        

File: 61-rotate-list/rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null || head.next==null ||k==0) return head;
        ListNode tail=head;
        int l=1;
        while(tail.next!=null){
          l++;
          tail=tail.next;
        }
        k=k%l;
        if(k==0) return head;
        tail.next=head;
        int n=l-k;
        ListNode temp=head;
       for(int i=1;i<n;i++){
        temp=temp.next;
       }
       ListNode nhead=temp.next;
       temp.next=null;
      
        return nhead;
    }
 
}